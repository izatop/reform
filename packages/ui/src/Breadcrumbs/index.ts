export * from "./BreadcrumbsProvider";
export * from "./BreadcrumbsState";
export * from "./Breadcrumbs";
export * from "./functions";
