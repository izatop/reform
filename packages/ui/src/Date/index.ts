export * from "./Calendar";
export * from "./DateRangePicker";
export * from "./functions";
export * from "./interfaces";
