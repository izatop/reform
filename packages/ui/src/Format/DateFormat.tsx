import * as React from "react";

interface IDateFormat {
    value: number | string | Date;
}

export const DateFormat: React.FC<IDateFormat> = (props) => (
    <>{new Date(props.value).toLocaleDateString()}</>
);
