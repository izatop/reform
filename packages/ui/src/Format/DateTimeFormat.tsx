import * as React from "react";

interface IDateFormat {
    value: number | string | Date;
    format?: string;
}

export const DateTimeFormat: React.FC<IDateFormat> = (props) => {
    const date = new Date(props.value);
    const format = props.format || "? ?";
    const value = format.replace("?", date.toLocaleDateString())
        .replace("?", date.toLocaleTimeString());

    return <>{value}</>;
};

DateTimeFormat.defaultProps = {format: "? ?"};
