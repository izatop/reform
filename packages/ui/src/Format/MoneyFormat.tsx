import * as React from "react";
import {MoneyFormatContext} from "./MoneyFormatContext";

interface INumberFormatProps {
    value: number | string;
    sign?: string;
}

export const MoneyFormat: React.FC<INumberFormatProps> = (props) => {
    const format = React.useContext(MoneyFormatContext);
    return <>{format.format(+props.value)}</>;
};
