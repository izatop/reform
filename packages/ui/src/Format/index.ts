export * from "./DateFormat";
export * from "./DateTimeFormat";
export * from "./TimeFormat";
