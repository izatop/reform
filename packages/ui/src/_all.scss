@import "~@reform/bulma/sass/variables";
@import "./Date/all";
