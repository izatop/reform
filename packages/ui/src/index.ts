export * from "./Accesibility";
export * from "./Table";
