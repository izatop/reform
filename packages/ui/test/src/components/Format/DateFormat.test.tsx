import renderer from "react-test-renderer";
import {DateFormat, DateTimeFormat, TimeFormat} from "../../../../src/Format";

const date = "2020-01-01T21:00:00.000Z";

test("Date and Time format", () => {
    expect(renderer.create(<DateFormat value={date}/>)).toMatchSnapshot();
    expect(renderer.create(<TimeFormat value={date}/>)).toMatchSnapshot();
    expect(renderer.create(<DateTimeFormat value={date}/>)).toMatchSnapshot();
    expect(renderer.create(<DateTimeFormat format={"Date: ?, Time: ?"} value={date}/>)).toMatchSnapshot();
});
