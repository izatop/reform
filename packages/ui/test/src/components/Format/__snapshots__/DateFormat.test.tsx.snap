// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date and Time format 1`] = `"1/2/2020"`;

exports[`Date and Time format 2`] = `"12:00:00 AM"`;

exports[`Date and Time format 3`] = `"1/2/2020 12:00:00 AM"`;

exports[`Date and Time format 4`] = `"Date: 1/2/2020, Time: 12:00:00 AM"`;
